src/app/shared/services/json-utils.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable } from '@angular/core';
+import { SchemaKeysStoreService } from './schema-keys-store.service';
+import * as _ from 'lodash';
+
+@Injectable()
+export class JsonUtilsService {
+
+  constructor(public schemaKeysStoreService: SchemaKeysStoreService) { }
+
+  filterJsonArray(jsonArray: Array<object>, filterExpression: string) {
+    let _jsonArray = [];
+    jsonArray.forEach(record => {
+      // cast to array for now before adding multiple filter expressions
+      let res = this.filterJson(record, [filterExpression]);
+      if (!_.isEmpty(res)) {
+        _jsonArray.push(res);
+      }
+    });
+    return _jsonArray;
+  }
+
+  filterJson(json: object, paths: Array<string>): object {
+    let result = {};
+    paths.forEach(path => {
+      let res = this.filterJsonRecursively(Object.assign({}, json), path);
```

Comment:
why `assign` here?, also if it's needed push it down the function that you're calling.

Reply:
`res` ?